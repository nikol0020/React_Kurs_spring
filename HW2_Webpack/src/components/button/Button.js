import React from 'react';
import './Button.scss';

import PropTypes from 'prop-types';

const Button = ({value, active, onClick}) => (
    <input
        type="button"
        className={active ? 'button button--active' : 'button'}
        value={value}
        onClick={onClick}
    />
);

Button.propTypes = {
    value: PropTypes.string,
    active: PropTypes.boolean,
    onClick: PropTypes.func
};


export default Button;
