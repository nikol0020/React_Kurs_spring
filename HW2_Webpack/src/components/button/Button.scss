.button {
  width: 70px;
  height: 30px;
  border-radius: 5px;
  background-color: #ccc;

  &:hover {
    cursor: pointer;
  }

  &__input {
    width: 100%;
    height: 30px;
    margin: 10px auto;
    border-radius: 5px;
  }

  &__buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
  }

  &--active {
    background-color: #a3a;
    color: #ccc;
    outline: 2px solid #aaa;
    border-radius: 0;
  }

}
