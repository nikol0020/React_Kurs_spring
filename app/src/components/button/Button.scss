.button {
  width: 70px;
  height: 30px;
  border-radius: 5px;
  background-color: #ccc;
  z-index: 1;

  &:hover {
    cursor: pointer;
  }

  &__input {
    width: 100%;
    height: 30px;
    margin: 10px auto;
    border-radius: 5px;
  }

  &__buttons {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
  }

  &--active {
    background-color: #a3a;
    color: #ccc;
    outline: 2px solid #aaa;
    border-radius: 0;
  }

  &__sort {
    width: 110px;;
  }

}
