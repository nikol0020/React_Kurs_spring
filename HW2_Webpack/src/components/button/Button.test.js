import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

describe('<Button /> shallow rendering', () => {

    it('matches the snapshot Button', () => {
        const wrapper = shallow(<Button />);
        expect(wrapper).toMatchSnapshot();
    });

});
