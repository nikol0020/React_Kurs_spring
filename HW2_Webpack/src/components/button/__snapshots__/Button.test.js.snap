// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shallow rendering matches the snapshot Button 1`] = `
<input
  className="button"
  type="button"
/>
`;
